### ISSUE TITLE
fatal error: concurrent map writes 

### ISSUE THREAD

**Issue by rueian** [blendle/zapdriver#9]
https://github.com/blendle/zapdriver/blob/d2d292b2ee7202aedceb5618bb7b49ce38acb306/core.go#L71

Hi, I found this line will print out `fatal error: concurrent map writes`
Maybe it should be wrap with a lock?

**Comment by jrbarron** [comment]
I also get this error quite often.

It might be as simple as using a [sync.Map](https://golang.org/pkg/sync/#Map) on this line:

https://github.com/blendle/zapdriver/blob/d2d292b2ee7202aedceb5618bb7b49ce38acb306/label.go#L43

**Comment by JeanMertz** [comment]
Thank you for reporting this. I've created a (admittedly rough) fix for this at #11.

I validated correct workings locally, but there's no CI setup for this project yet, so I'm going to merge and release this. Feel free to let me know if there are still issues.

**Comment by lokeshj** [comment]
I'm getting the same error on 1.1.4

```
fatal error: concurrent map writes

goroutine 10 [running]:
runtime.throw(0xf49c84, 0x15)
	/usr/local/go/src/runtime/panic.go:608 +0x72 fp=0xc0002315d0 sp=0xc0002315a0 pc=0x42db62
runtime.mapassign_faststr(0xddb4c0, 0xc0003b4fc0, 0xc0002780a0, 0x9, 0xc000248030)
	/usr/local/go/src/runtime/map_faststr.go:275 +0x3bf fp=0xc000231638 sp=0xc0002315d0 pc=0x4143af
github.com/blendle/zapdriver.(*core).With(0xc0003154c0, 0x17ee1f0, 0x0, 0x0, 0xd9fba0, 0xc000278001)
	/home/lokesh/go/pkg/mod/github.com/blendle/zapdriver@v1.1.4/core.go:47 +0x150 fp=0xc000231718 sp=0xc000231638 pc=0x6d6f70
go.uber.org/zap.(*Logger).With(0xc0002f5920, 0xc0004a6040, 0x1, 0x1, 0x9)
	/home/lokesh/go/pkg/mod/go.uber.org/zap@v1.9.1/logger.go:164 +0xad fp=0xc000231760 sp=0xc000231718 pc=0x6cfbdd
```
Its coming from the `With` method
https://github.com/blendle/zapdriver/blob/87f0bcb9f9e50b5423581426cc576b6ced81482d/core.go#L47

**Comment by JeanMertz** [comment]
Thanks @lokeshj, I indeed did not lock all the mutexes correctly. This should be fixed in #12. I'll also cut a new release with that fix applied.
